testng-core/src/main/java/org/testng/SuiteRunner.java

```diff
@@ -189,25 +192,20 @@ private void init(
           };
     }
     // Add our own IInvokedMethodListener
-    invokedMethodListeners = Maps.newConcurrentMap();
-    if (invokedMethodListener != null) {
-      for (IInvokedMethodListener listener : invokedMethodListener) {
-        invokedMethodListeners.put(listener.getClass(), listener);
-      }
+    invokedMethodListeners = Maps.synchronizedLinkedHashMap();
+    for (IInvokedMethodListener listener :
+        Optional.ofNullable(invokedMethodListener).orElse(Collections.emptyList())) {
```

Comment:
invokedMethodListener is not nullable, it was just created one line before.

Reply:
Not true. Please check the variable names. We are doing a null check on `invokedMethodListener` using `Optional.ofNullable()`, but we are initialising a different variable `invokedMethodListeners` (this is a plural)

Reply:
Oops, right! :)